### PR TITLE
Encrypt the password

### PR DESCRIPTION
**Pull request by wlngo**
encrypt shenyu-admin login password #5255

### CODE REVIEW

**Comment by moremind** [comment]
do you test this pr?

**Comment by 2252603132** [comment]
yes in h2 database

**Comment by moremind** [comment]
add my wechat: yijia1949

**Comment by 2252603132** [comment]
I have added

**Comment by li-keguo** [comment]
This topic also involves client meta-information registration login.

**Comment by 2252603132** [comment]
> 本主题还涉及客户端元信息注册登录。
Can you be more specific?

**Comment by li-keguo** [comment]
> > 本主题还涉及客户端元信息注册登录。
> > Can you be more specific?

see:
https://github.com/apache/shenyu/blob/master/shenyu-register-center/shenyu-register-client/shenyu-register-client-http/src/main/java/org/apache/shenyu/register/client/http/HttpClientRegisterRepository.java

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/DashboardUserServiceImpl.java [hunk 1]
```diff
@@ -319,7 +337,25 @@ public boolean checkUserPassword(final String userId) {
...
+            String key = secretProperties.getKey();
+            String iv = secretProperties.getIv();
+            Cipher cipher = Cipher.getInstance("AES/CTR/NoPadding");
+            SecretKey secretKey = new SecretKeySpec(key.getBytes(StandardCharsets.UTF_8), "AES");
+            IvParameterSpec ivParameterSpec = new IvParameterSpec(iv.getBytes(StandardCharsets.UTF_8));
+            cipher.init(Cipher.DECRYPT_MODE, secretKey, ivParameterSpec);
+            byte[] decryptedBytes = cipher.doFinal(Base64.getDecoder().decode(password));
+            encodedPassword = new String(decryptedBytes);
+        } catch (Exception e) {
+            LOG.error("AES decode error", e);
```
Comment (moremind) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/DashboardUserServiceImpl.java:353]: throw exception include message

**Review by moremind — CHANGES_REQUESTED** [review]

**Comment by moremind** [comment]
the password of the database should be encrypted

**Comment by li-keguo** [comment]
> the password of the database should be encrypted

The database password change needs to be compatible with upgrade and downgrade scenarios. I think another topic can be used to change the database password.

**Review by li-keguo — APPROVED** [review]

**Review by li-keguo — APPROVED** [review]

**Comment by moremind** [comment]
need to change

File: shenyu-register-center/shenyu-register-client/shenyu-register-client-http/src/main/java/org/apache/shenyu/register/client/http/HttpClientRegisterRepository.java [hunk 2]
```diff
@@ -152,6 +158,8 @@ public void closeRepository() {
         }
     }
 
+
```
Comment (moremind) [inline shenyu-register-center/shenyu-register-client/shenyu-register-client-http/src/main/java/org/apache/shenyu/register/client/http/HttpClientRegisterRepository.java:161]: remove this

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-register-center/shenyu-register-client/shenyu-register-client-http/src/main/java/org/apache/shenyu/register/client/http/HttpClientRegisterRepository.java (hunk 2 above)
Comment (wlngo) [reply shenyu-register-center/shenyu-register-client/shenyu-register-client-http/src/main/java/org/apache/shenyu/register/client/http/HttpClientRegisterRepository.java:161]: 好的

**Review by moremind — APPROVED** [review]

**Pull request merged**